server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -291,6 +304,43 @@ long getAbsoluteStartMillis() {
         long buildTookInMillis() {
             return TimeUnit.NANOSECONDS.toMillis(relativeCurrentNanosProvider.getAsLong() - relativeStartNanos);
         }
+
+        public void setPhaseTookEnabled(boolean phaseTookEnabled) {
+            this.phaseTookEnabled = phaseTookEnabled;
+        }
+
+        public boolean isPhaseTookEnabled() {
+            return phaseTookEnabled;
+        }
+
+        SearchResponse.PhaseTook getPhaseTook() {
+            if (phaseTookEnabled) {
+                Map<String, Long> phaseTookMap = new HashMap<>();
+                // Convert Map<SearchPhaseName, Long> to Map<String, Long> for SearchResponse()
+                for (SearchPhaseName searchPhaseName : SearchPhaseName.values()) {
+                    phaseTookMap.put(searchPhaseName.getName(), phaseStatsMap.get(searchPhaseName));
+                }
+                return new SearchResponse.PhaseTook(phaseTookMap);
+            } else {
+                return SearchResponse.PhaseTook.NULL;
```

Comment:
> Otherwise we cannot tell the difference between EMPTY and non-empty with took times all zero.

Why do we need to determine this? To decide whether we need to add this as part of response or not?

Reply:
Yes that is correct

Reply:
Then why not just use an empty map(map.isEmpty()) to determine that. Why do we need to initialize it all with values?
Don't initialise phaseStatsMap when it is not needed in response?